### PR TITLE
add: github action corn sync data

### PR DESCRIPTION
**Pull request by yihong0618**
Please change the secret token username and email in GitHub action.

### CODE REVIEW

**Comment by yihong0618** [comment]
@antfu 
Try to add feature for #4

**Comment by antfu** [comment]
Thanks for working on this! However, I don't think committing the entire data.json is a good approach, think about just committing the head sha of the source repo would be enough (netlify will handle the fetch and build process)

File: .github/workflows/run_cafe_sync.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+      - name: Push new Cafe
+        uses: github-actions-x/commit@v2.6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: "Add new Cafe"
+          files: src/data.json
+          rebase: 'true'
+          name: yihong0618
+          email: zouzou0208@gmail.com
```
Comment (antfu) [inline .github/workflows/run_cafe_sync.yml]:
```suggestion
      - uses: EndBug/add-and-commit@v4
        with:
          message: 'chore: update source'
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

And I would suggest using this action instead :)

Comment (yihong0618) [reply .github/workflows/run_cafe_sync.yml]: Cool, I think you are right, thanks.  That will be easily. So can I create one `json file` only store the commit-sha on source repo if it change we trigger the action commit and netlify will do the other job?

Comment (antfu) [reply .github/workflows/run_cafe_sync.yml]: Exactly! I think a plain text file with the sha will do. You can run this line to get the latest sha

```bash
curl https://api.github.com/repos/ElaWorkshop/awesome-cn-cafe/branches/master --silent | grep "sha" | head -n1 > sha
```

File: .github/workflows/run_cafe_sync.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+      
+      - name: Get sha
+      - run: curl https://api.github.com/repos/ElaWorkshop/awesome-cn-cafe/branches/master --silent | grep "sha" | head -n1 > src/sha 
+
+      - name: Push new Cafe
+      - uses: EndBug/add-and-commit@v4
+        with:
+          message: 'chore: update source'
```
Comment (antfu) [inline .github/workflows/run_cafe_sync.yml]:
```suggestion
          message: "chore: update source sha"
```

**Comment by antfu** [comment]
Looking great to me! Thanks for your work, merging now :)

**Comment by yihong0618** [comment]
> Looking great to me! Thanks for your work, merging now :)

Thank you, really learned a lot from your kind advice and repo.

**Pull request merged**